build-system/tasks/dist.js

```diff
@@ -51,16 +51,52 @@ const {compileCss, cssEntryPoints} = require('./css');
 const {createCtrlcHandler, exitCtrlcHandler} = require('../ctrlcHandler');
 const {isTravisBuild} = require('../travis');
 const {maybeUpdatePackages} = require('./update-packages');
+const tempy = require('tempy');
+const conf = require('../build.conf');
+const {isCommonJsModule} = require('../compile/compile-utils');
 
 const {green, cyan} = colors;
 const argv = require('minimist')(process.argv.slice(2));
 
+
+const deglob = require('globs-to-files');
+const babel = require("@babel/core");
+
+const allJsSrcsGlob = [
+  '3p/**/*.js',
+  'ads/**/*.js',
+  'extensions/**/*.js',
+  'src/**/*.js',
+];
+
+function transferSrcsToTempDir() {
+  const tmp = `${tempy.directory()}/amphtml`;
+  process.env.AMP_TMP_DIR = tmp;
+  console.log('new tmp dir', tmp);
+  const files = deglob.sync(allJsSrcsGlob);
+  console.log('Executing babel transforms');
+  files.forEach(file => {
+    const {code} = babel.transformFileSync(file, {
+      plugins: conf.plugins({
+        isEsmBuild: argv.esm,
+        isCommonJsModule: isCommonJsModule(file),
+        isForTesting: argv.fortesting,
+      }),
+      retainLines: true,
+    });
+    const name = `${tmp}${file.replace(process.cwd(), '')}`;
+    fs.outputFileSync(name, code);
```

Comment:
Sourcemaps, too.

Reply:
since this is just the preprocess step, we shouldnt need sourcemaps here. we'll still want to rely on CC for the sourcemaps.

Reply:
LMK, if my statement is correct. Also PTAL when you can. Thanks!